### PR TITLE
feature: Enable client certificate rotation by using tls.Config@GetClientCertificate

### PR DESCRIPTION
**Pull request by benkelukas**
Enable client certificate rotation by providing a callback to tls.Config@GetClientCertificate when setting up Temporal connection

# Reason for This PR

Closes: #522 

## Description of Changes

Added a callback for GetClientCertificate to enable client certificate rotation when mTLS connection fails, e.g. after it changes in the referenced cert and key files

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the MIT license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: tls.go [hunk 1]
```diff
@@ -39,23 +33,30 @@ func initTLS(cfg *Config) (*tls.Config, error) {
...
-		MinVersion:   tls.VersionTLS12,
-		Certificates: []tls.Certificate{cert},
+		ServerName:           cfg.TLS.ServerName,
+		MinVersion:           tls.VersionTLS12,
+		GetClientCertificate: getClientCertificate(cfg),
 	}, nil
 }
+
+// getClientCertificate is used for tls.Config struct field GetClientCertificate and enables re-fetching the client certificates when they expire
+func getClientCertificate(cfg *Config) func(info *tls.CertificateRequestInfo) (*tls.Certificate, error) {
```
Comment (rustatian) [inline tls.go:53]: `info` is unused, should be _, or removed

Comment (rustatian) [reply tls.go]: I meant, both `infos` 😃 They're both unused.

Comment (benkelukas) [reply tls.go]: sorry! Missed that one 🤦 fixed

**Comment by rustatian** [comment]
Thanks, @benkelukas 👍 LGTM, will be released next week with RR v2024.1.3.

**Review by rustatian — APPROVED** [review]

**Comment by benkelukas** [comment]
Thanks @rustatian - that's great news! 🎉

**Pull request merged**